### ISSUE TITLE
[Bug Report] Softmax fails due to L1 OOM

### ISSUE THREAD

**Issue by rfurko-tt** [tenstorrent/tt-metal#20099]
**Describe the bug**
```
import torch
import ttnn

with ttnn.manage_device(device_id=0) as device:
    x_torch = torch.ones((2, 1, 16384, 8192), dtype=torch.bfloat16)
    x = ttnn.from_torch(x_torch, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)
    compute_kernel_config = ttnn.WormholeComputeKernelConfig(
        math_fidelity=ttnn.MathFidelity.HiFi4,
        math_approx_mode=False,
        fp32_dest_acc_en=True,
        packer_l1_acc=False,
    )
    # two commented options results in crash due to `numeric_stable` parameter
    # y = ttnn.softmax(x, -1, compute_kernel_config=compute_kernel_config, numeric_stable=True)
    # y = ttnn.softmax(x, -1, numeric_stable=True)
    y = ttnn.softmax(x, -1, numeric_stable=False)
    print('y0:', y.shape)
```

**To Reproduce**
* Uncomment one of the commented ttnn calls 

**Expected behavior**
Should not be L1 OOM, as it works with `numeric_stable` option off.

**Please complete the following environment information:**
- OS: Ubuntu 20.04
- Version of software: main

**Additional context**
The issue is kernel specific. For some reason if `numeric_stable` is turned on we assume that whole inner dimension fits in L1 memory. 

Example code: `uint32_t in0_t = numeric_stable ? tt::div_up(Wt, block_size) * block_size : block_size * 2;`

**Comment by rfurko-tt** [comment]
@dmakoviichuk-tt FYI

**Comment by cmaryanTT** [comment]
This is likely a duplicate of a known class of issues.

**Comment by bbradelTT** [comment]
@vsureshTT is working on a fix.
